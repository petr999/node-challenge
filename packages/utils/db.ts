import { Client } from 'pg';
import config from 'config';
import { Connection } from 'typeorm';

export const dbConnection: {
  typeOrm?: Connection
  [k: string]: any
} = {};

export function connect() {
  dbConnection.client = new Client(config.db);
  return dbConnection.client.connect();
}

export async function query(queryString: string, parameters?: any) {
  try {
    if (!dbConnection.client) await connect();
  } catch (e) {
    dbConnection.client = undefined;
    throw e;
  }

  return dbConnection.typeOrm.query(queryString, parameters);
}
