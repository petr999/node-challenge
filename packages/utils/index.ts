export * from './formatter';
export * from './dal';
export * from './errors';
