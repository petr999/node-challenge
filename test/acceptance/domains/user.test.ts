import { Api } from '../utils/api';

describe('Unknown user not found', () => {
  test('Random user returns Not Found', (done) => {
    Api.get('/user/v1/get-user-details?userId=12992732-bd00-4a9a-9426-131bd618740e')
      .expect(404, done);
  });
});

describe('User found in database', () => {
  test('User known to be contained in database got fetched', (done) => {
    Api.get('/user/v1/get-user-details?userId=da140a29-ae80-4f0e-a62d-6c2d2bc8a474')
      .expect(200)
      .end((err, res) => {
        expect(err).toBeFalsy();
        expect(res.body).toMatchSnapshot('user-details-by-id-da140a29-ae80-4f0e-a62d-6c2d2bc8a474');
        done();
      });
  });
});
