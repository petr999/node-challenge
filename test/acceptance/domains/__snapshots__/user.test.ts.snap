// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User found in database User known to be contained in database got fetched: user-details-by-id-da140a29-ae80-4f0e-a62d-6c2d2bc8a474 1`] = `"{\\"first_name\\":\\"Jeppe\\",\\"last_name\\":\\"Rindom\\",\\"company_name\\":\\"pleo\\"}"`;
